src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -900,9 +915,77 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         AddAction(*cmd, keys);
     }
 
-    IVector<Model::Command> ActionMap::FilterToSendInput(
-        winrt::hstring currentCommandline)
+    // Update ActionMap's cache of actions for this directory. We'll look for a
+    // .wt.json in this directory. If it exists, we'll read it, parse it's JSON,
+    // then take all the sendInput actions in it and store them in our
+    // _cwdLocalSnippetsCache
+    std::vector<Model::Command> ActionMap::_updateLocalSnippetCache(winrt::hstring currentWorkingDirectory)

```

Comment:
I'm actually fairly uncomfortable with this behavior living in SettingsModel, but oh well

Reply:
(it's where the json parser lives, so it makes sense)